editor.planx.uk/src/@planx/components/MultipleFileUpload/Public.tsx

```diff
@@ -70,6 +72,13 @@ function Component(props: Props) {
 
   const [slots, setSlots] = useState<FileUploadSlot[]>([]);
 
+  // Track number of slots, and open modal when this increases
+  const previousSlotCount = usePrevious(slots.length);
+  useEffect(() => {
+    if (previousSlotCount === undefined) return;
+    if (slots.length > previousSlotCount) setShowModal(true);
+  }, [slots.length]);
```

Comment:
nice use of `usePrevious` here :heavy_plus_sign: 

also didn't think of passing `[slots.length]` in the callback like this - super handy !